Framework/Algorithms/test/CalMuonDetectorPhasesTest.h

```diff
@@ -98,6 +112,34 @@ class CalMuonDetectorPhasesTest : public CxxTest::TestSuite {
     TS_ASSERT(!calc->isExecuted());
   }
 
+  /**
+   * Test that the algorithm can handle a WorkspaceGroup as input without
+   * crashing
+   * We have to use the ADS to test WorkspaceGroups
+   */
+  void testValidateInputsWithWSGroup() {
+    auto ws1 = boost::static_pointer_cast<Workspace>(
+        createWorkspace(2, 4, "Microseconds"));
+    auto ws2 = boost::static_pointer_cast<Workspace>(
+        createWorkspace(2, 4, "Microseconds"));
+    AnalysisDataService::Instance().add("workspace1", ws1);
+    AnalysisDataService::Instance().add("workspace2", ws2);
+    auto group = boost::make_shared<WorkspaceGroup>();
+    AnalysisDataService::Instance().add("group", group);
+    group->add("workspace1");
+    group->add("workspace2");
+    TestCalMuonDetectorPhases calc;
+    calc.initialize();
+    calc.setChild(true);
+    TS_ASSERT_THROWS_NOTHING(calc.setPropertyValue("InputWorkspace", "group"));
```

Comment:
I tried this and got "enter a name for the input in/out workspace" - it only seems to work if the workspaces are added to the ADS.